src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1285,9 +1284,17 @@ std::pair< bool, nuraft::cb_func::ReturnCode > RaftReplDev::handle_raft_event(nu
             sisl::VectorPool< repl_req_ptr_t >::free(reqs);
         }
         if (ret == nuraft::cb_func::ReturnCode::Ok) { m_state_machine->inc_next_batch_size_hint(); }
-        return {true, ret};
+        return ret;
+    }
+    case nuraft::cb_func::Type::BecomeFollower: {
+        become_follower_cb();
+        return nuraft::cb_func::ReturnCode::Ok;
+    }
+    case nuraft::cb_func::Type::JoinedCluster:
```

Comment:
is it on purpose that JoinedCluster will call become_leader_cb?

Reply:
sorry that implies become_follower,  changed the structure.